drivers/kscan/Kconfig

```diff
@@ -27,4 +27,77 @@ config KSCAN_INIT_PRIORITY
 	help
 	  Keyboard scan device driver initialization priority.
 
+config KSCAN_MATRIX
+	bool "keyboard scan matrix"
+	help
+	  A matrix circuit is used in most computer keyboards where the key
+	  switches are connected by a grid of wires, similar to a diode matrix.
+
+if KSCAN_MATRIX
+config KSCAN_MATRIX_MAX_COLUMNS
+	int "keyboard matrix max column"
+	default 18
+	help
+	  Maximum columns of the matrix circuit.
+
+config KSCAN_MATRIX_MAX_ROWS
+	int "keyboard matrix max row"
+	default 8
+	help
+	  Maximum rows of the matrix circuit.
+
+config KSCAN_MATRIX_DRIVER
+	bool "Keyboard matrix driver for scanning"
+	help
+	  Keyboard Scan Matrix Driver Support
+
+if KSCAN_MATRIX_DRIVER
+
+config KSCAN_MATRIX_DRV_NAME
+	string "Keyboard Scan Matrix Driver name"
+	default "kscan_matrix"
+
+config KSCAN_MATRIX_TASK_INIT_PRIORITY
+	int "Keyboard matrix task init priority"
+	default 95
+	help
+	  Keyboard matrix task initialization priority.
+config KSCAN_MATRIX_THREAD_PRIO
+	int "Keyboard matrix thread priority"
+	default 0
+	help
+	  Priority level for kscan matrix polling thread.
+
+config KSCAN_MATRIX_DEBOUNCE_DOWN
+	int "keyboard matrix debounce down"
+	default 10
+	help
+	  Determines the time in msecs for debouncing a key press.
+
+config KSCAN_MATRIX_DEBOUNCE_UP
+	int "keyboard matrix debounce up"
+	default 20
+	help
+	  Determines the time in msecs for debouncing a key release.
+
+config KSCAN_MATRIX_POLL_PERIOD
+	int "keyboard matrix poll period"
+	default 5
+	help
+	  Defines the poll period in msecs between between matrix scans
+
+config KSCAN_MATRIX_POLL_COL_OUTPUT_SETTLE_TIME
+	int "keyboard matrix poll column output settle time"
+	default 50
+	help
+	  Delay (us) between setting column output and waiting for it
+	  to settle
+
+endif # KSCAN_MATRIX_DRIVER
+
+comment "Keyboard Scan Matirx Chip Drivers"
+source "drivers/kscan/Kconfig.npcx"
```

Comment:
Move this with the other drivers up above.  Note that `if/endif` blocks in Kconfig files do not skip `source` statements.

https://docs.zephyrproject.org/latest/build/kconfig/tips.html#lack-of-conditional-includes

Reply:
Thanks for the reminder.
The original thinking to put source source "drivers/kscan/Kconfig.npcx" here is let the layout of npcx selection within the KSCAN_MATRIX scope.
Anyway, I have moved this line up, together with the other  source "xxx files".